🐛 Fixes for SPM

abegehr opened the pull request:
Had a few more issues with Package.swift and some swift code in Xcode:

1. ZipArchive only supports iOS ≥15.5
2. Library Name needs to match package name: "CapgoCapacitorUpdater" (and target attr references target bellow).
3. Needs to support more versions than just capacitor-swift-pm "6.0.0", instead: "^6.0.0".
4. Import "ZipArchive" seems to import SSZipArchive, removed second camImport declaration.
5. UIKit import missing, required for UIDevice.
6. Some Syntax issues with Version not being available as string, probably due to newer platform version.

I've tested it and it works in my project.

Some screenshots of the errors, pre-fix:

<img width="1284" alt="Screenshot 2025-01-06 at 19 20 18" src="https://github.com/user-attachments/assets/98742399-25db-4a0b-92ca-499f6a32cefd" />
<img width="998" alt="Screenshot 2025-01-06 at 19 22 48" src="https://github.com/user-attachments/assets/73aaf9da-4f4e-4cd3-82c4-13e0906a1ca9" />
<img width="353" alt="Screenshot 2025-01-06 at 19 24 05" src="https://github.com/user-attachments/assets/28cde0e0-4ed5-491e-8450-3c3e9b7c855d" />
<img width="1656" alt="Screenshot 2025-01-06 at 19 29 44" src="https://github.com/user-attachments/assets/e5ab3607-8aac-4339-8cbc-47ee52093eba" />

abegehr commented:
Coming from #510

riderx commented:
We will not upgrade the version minimum ios, ee have to follow capacitor one, so better to lock the version of ZipArchive to a version who doesn’t require the bump

riderx commented:
I just realized all version below 2.5 are not available in spm, so we can’t do anything for now for this

riderx commented:
I did raise issue, our ask is quite nonsense and i would understand the creator of the lib refuse.
Let see

marius-se commented:
What if we replace ZipArchive with a different Swift package that can zip/unzip?

riderx commented:
ZipArchive has been used since day one in this plugin, we cannot change that so easily.
Many step are depending on it, including checksum etc.
in v7 we could do that in favor of Brotli, who is already used for partial

marius-se commented:
> ZipArchive has been used since day one in this plugin, we cannot change that so easily. Many step are depending on it, including checksum etc. in v7 we could do that in favor of Brotli, who is already used for partial

I just tried removing it (with [Zip](https://github.com/marmelroy/Zip)) and I only had to replace a single call to ZipArchive in order to compile the package. I could open a PR for that, based on @abegehr 's branch if you want @riderx
No need to bump any minimum iOS version then and no breaking changes. Let me know. We're one of your paying customers and would love to get this upstream :)

riderx commented:
@marius-se I wish that to be so easy. Changing a lib used by millions of devices daily has a huge impact.
You can make a PR if you find one lib who fit, sadly for this one I think it's a dead end:
not maintained for 11 months:
https://github.com/marmelroy/Zip/issues/256
Do not incorporate privacy manifest (who is mandatory to any new release):
https://github.com/marmelroy/Zip/issues/259

I will do some more research as well

riderx commented:
I think i found the solution to force the version in SPM

riderx commented:
psuhed

riderx commented:
Please try [6.12.1](https://github.com/Cap-go/capacitor-updater/tree/refs/tags/6.12.1)

riderx commented:
Sorry, I found another issue, try [6.12.3](https://github.com/Cap-go/capacitor-updater/tree/refs/tags/6.12.3)
Version package was linked to an incorrect package in SPM

abegehr commented:
awesome, works! thanks @riderx

abegehr commented:
I've updated this PR to include fixes from 6.12.3:

1. ZipArchive only supports iOS ≥15.5 – fixed by @riderx by pinning version ✅
2. Library Name needs to match package name: "CapgoCapacitorUpdater" – fix is in this PR ☑️
3. Needs to support more versions than just capacitor-swift-pm "6.0.0", instead: "^6.0.0" – fix is in this PR ☑️
4. Import "ZipArchive" imports SSZipArchive, removed second camImport declaration  – fix is in this PR ☑️
5. UIKit import missing, required for UIDevice  – fix is in this PR ☑️
6. Some Syntax issues with Version not being available as string, probably due to newer platform version - Version dependency link updated ✅

@riderx, please let me know if you agree with those fixes.

riderx commented:
For the package version of capacitor swift pm we should follow the template:
https://github.com/ionic-team/create-capacitor-plugin/blob/433a3013d824308e9649f2dfc44e5ba349e3553b/assets/plugin-template/Package.swift.mustache
Instead something custom not recommended, outside of this i can accept

riderx commented:
Can you fix it and i accept your pr?

abegehr commented:
yes, doing now, thanks @riderx

abegehr commented:
🙌
